### DISCUSSION TITLE
Chat Survey not appearing after closing chat

### DISCUSSION THREAD

**Discussion by Maddn3ss** [LiveHelperChat/livehelperchat#2016] [Q&A]
Users reported that survey no longer appear after chat agent closes their chat.

Tested on versions: v3.29, v4.15 and latest v4.29
Last time it was working: prior to v3.29

After series of testing, it was noticeable that for any new chat user that initiates a chat session, the survey does not appear upon chat closure. However, once an agent manually redirects the user to the survey and closes the chat session, the survey will then appear at the end of the closed chat session and moving forward all new chat sessions from the same user on the same browser page. Unfortunately, this will all reset once the customer closes the browser page from where the chat widget is applied and the same customer is treated as a new chat user and survey will not appear when chat is closed.

Survey form is applied under Department > Survey
System configuration > Online Tracking enabled

Could there be missing permissions or configuration ?

**Comment by remdex** [comment]
To hard to tell as there is no such issue. You can try to reproduce on demo or official hosting. If you manage todo so then I could take a look. At the moment I tested myself and all works as expected. Try first to do that without having any widget theme assigned.

**Comment by Bucky101** [comment]
Hey, 
I have reproduced the issue on the demo system using https://livehelperchat.com/lhcsamples/sample-2.html via a JS embedded widget.
On our system, I can confirm its the same behavior.  If we initiate a chat via the https://demo.livehelperchat.com/ type UI it works as expected and we are auto-transferred to the survey, but not via the widget.

**Comment by Antonamy** [comment]
hi, 

I just saw this and reported this last month on discord but could not figure it out as well and I could not provided need data to Remi.

![image](https://github.com/LiveHelperChat/livehelperchat/assets/150763687/6999ed5b-2b80-4252-965d-bd1beaa65c8a)

**Reply by remdex** [reply]
Guys!

If none of you can provide steps to reproduce how I'm supposed to fix anything???

You can always order automated hosting it's free for two weeks make your reproduction there and send me link to embed code and back office link.

It's just disrespectful to say something does not work and do not give any clue what you have there... App development does not work like that I'm afraid...

**Comment by remdex** [comment]
I saw that @Bucky101 finally managed reproduce, so I'll take a look there.

**Reply by remdex** [reply]
Sorry, but it does work here.
![image](https://github.com/LiveHelperChat/livehelperchat/assets/1146085/5eda1475-224a-4ea5-8219-c6256a049448)

Read more. So still valid steps to reproduce is required.
https://doc.livehelperchat.com/docs/chat/survey#i-close-a-chat-but-i-dont-see-any-survey

**Comment by wasimtikki120** [comment]
The issue you described seems to be related to the survey functionality in a chat application, specifically, that the survey is not consistently appearing for all users after a chat session is closed. While I can't provide specific details about your system's configuration, I can suggest some general troubleshooting steps and areas to investigate:

1. **Check Permissions:**
   - Ensure that the necessary permissions are granted for users to access and submit surveys. Permissions might have changed in the versions you mentioned.

2. **Survey Configuration:**
   - Review the survey configurations in the Department settings. Ensure that the survey is set up to appear after the chat is closed. Check if there are any conditions or triggers for survey display.

3. **Online Tracking:**
   - Since you mentioned that online tracking is enabled, make sure that it's configured correctly. Verify if the online tracking feature might be affecting the behavior of the survey.

4. **Browser Storage or Cookies:**
   - The behavior you described, where the survey appears after an agent manually redirects the user but not when the customer closes the browser, could be related to browser storage or cookies. Check if the survey relies on browser storage or cookies for tracking.

5. **Session Management:**
   - Investigate how user sessions are managed in your application. If sessions are not being maintained correctly, it might explain why the survey works for returning users but not for new sessions.

6. **Check for Updates:**
   - Ensure that you are using the latest version of your chat application. Check for any updates or patches that might address issues related to survey functionality.

7. **Review Release Notes:**
   - Review the release notes for the versions you tested. Developers often document changes, bug fixes, and new features in release notes. Look for any information related to survey behavior.

8. **Logs and Error Messages:**
   - Check logs and error messages generated by your chat application. Look for any errors or warnings that might indicate issues with survey functionality.

9. **Testing Environment:**
   - Consider setting up a controlled testing environment where you can reproduce the issue consistently. This can help you debug and identify the root cause more effectively.

10. **Contact Support:**
    - If you are unable to identify the issue, consider reaching out to the support or community forum for the chat application. Provide details about your configuration, steps to reproduce the issue, and any error messages encountered.

Remember, the specifics of troubleshooting may vary based on the architecture and design of your chat application. It's recommended to involve your development or technical support team to assist in identifying and resolving the issue.
